code/game/turfs/simulated/floor/asteroid.dm

```diff
@@ -31,79 +47,42 @@
 /turf/simulated/floor/plating/asteroid/remove_plating()
 	return
 
-/turf/simulated/floor/plating/asteroid/ex_act(severity, target)
-	switch(severity)
-		if(3)
-			return
-		if(2)
-			if(prob(20))
-				gets_dug()
-		if(1)
-			gets_dug()
-
-/turf/simulated/floor/plating/asteroid/attackby(obj/item/W, mob/user, params)
+/turf/simulated/floor/plating/asteroid/attackby(obj/item/I, mob/user, params)
 	//note that this proc does not call ..()
-	if(!W || !user)
-		return 0
+	if(!I || !user)
+		return FALSE
 
-	if((istype(W, /obj/item/shovel) || istype(W, /obj/item/pickaxe)))
-		var/turf/T = get_turf(user)
-		if(!istype(T))
-			return
+	if((istype(I, /obj/item/shovel) || istype(I, /obj/item/pickaxe)))
+		if(!can_dig(user))
+			return TRUE
 
-		if(dug)
-			to_chat(user, "<span class='warning'>This area has already been dug!</span>")
+		if(!isturf(user.loc))
 			return
 
 		to_chat(user, "<span class='notice'>You start digging...</span>")
-		playsound(src, W.usesound, 50, 1)
-		if(do_after(user, 20 * W.toolspeed, target = src))
+
+		playsound(src, I.usesound, 50, 1)
+		if(do_after(user, 20 * I.toolspeed, target = src))
+			if(!can_dig(user))
+				return TRUE
 			to_chat(user, "<span class='notice'>You dig a hole.</span>")
-			gets_dug()
-			return
+			getDug()
+			return TRUE
 
-	else if(istype(W,/obj/item/storage/bag/ore))
-		var/obj/item/storage/bag/ore/S = W
+	else if(istype(I, /obj/item/storage/bag/ore))
+		var/obj/item/storage/bag/ore/S = I
 		if(S.collection_mode == 1)
 			for(var/obj/item/stack/ore/O in src.contents)
-				O.attackby(W,user)
-				return
-
-	else if(istype(W, /obj/item/stack/tile))
```

Comment:
I think we'll want to keep this. As our turf system is vastly outdated, this is necessary to make it possible to build bases on Lavaland with normal turfs.

Reply:
Ahh, thanks for letting me know.